agent/utils/mongo_fix/mongo_fix.go

```diff
@@ -40,6 +40,10 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) {
 	if username != "" || password != "" {
 		clientOptions.Auth.Username = username
 		clientOptions.Auth.Password = password
+
+		// set this flag to connect to arbiter when there authentication is enabled
+		b := true
+		clientOptions.AuthenticateToAnything = &b //nolint:staticcheck
```

Comment:
we have `pointer` library for this kind of stuff

Reply:
thanks, fixed